reporting/ReportingObserverOptions-types.html

```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE HTML>
+<meta charset=utf-8>
+<title>ReportingObserverOptions: types member</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+  test(() => {
+    const expectedException = {}
+    assert_throws_exactly(expectedException, () => {
+      new ReportingObserver(() => {}, { types: { *[Symbol.iterator]() { throw expectedException; } } });
+    });
+  }, "iterator that throws in `types`");
+
+  test(() => {
+    const expectedException = {}
+    assert_throws_exactly(expectedException, () => {
+      new ReportingObserver(() => {}, { types: [ { valueOf: () => { throw expectedException; } } ] });
+    });
+  }, "object that throws when getting `valueOf` in `types` array");
```

Comment:
This test fails in Chrome Canary and Firefox Nightly. I'm not sure why. Do they not iterate the array during webidl argument conversion as per https://heycam.github.io/webidl/#create-sequence-from-iterable ? Or is the test incorrect?

Reply:
@EdgarChen @TimothyGu can you help here?

Reply:
Why should this throw an exception? `types` is a `sequence<DOMString>`, so the object `({ valueOf: () => { throw expectedException; } })` will be coerced into a DOMString. Coercing an object into a string calls `toString` before `valueOf` (see https://tc39.es/ecma262/#sec-ordinarytoprimitive), and `Object.prototype.toString` returns `[object Object]`.

I'm not sure what you are trying to test here, but here are a few options to generate exceptions as expected:
```js
new ReportingObserver(() => {}, { types: [ Object.create(null) ] });
new ReportingObserver(() => {}, { types: [ { toString: () => { throw expectedException; } } ] });
new ReportingObserver(() => {}, { types: [ { toString: undefined, valueOf: () => { throw expectedException; } } ] });
new ReportingObserver(() => {}, { types: [ { [Symbol.toPrimitive]: () => { throw expectedException; } } ] });
```

Reply:
For what it's worth, WebKit seems to agree with Chromium and Firefox here.